client/domain/PerformanceTestObjectFactory.js

```diff
@@ -0,0 +1,42 @@
+// Copyright 2017 The TIE Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating new frontend instances of PerformanceTest
+ * domain objects.
+ */
+
+tie.factory('PerformanceTestObjectFactory', [
+  function() {
+    var PerformanceTest = function(performanceTestDict) {
+      this._inputDataAtom = performanceTestDict.input_data_atom;
+      this._transformationFunction =
+        performanceTestDict.transformation_function;
+      this._expectedPerformance = performanceTestDict.expected_performance;
+      this._evaluationFunction = performanceTestDict.evaluation_function;
+    };
+
+    // Instance methods.
+    PerformanceTest.prototype.getTransformationFunction = function() {
```

Comment:
Discussed with @eyurko; this is a no-op. We plan to add other functions lazily to the domain objects as we need them, in order to minimize the amount of boilerplate and to make it less work to handle a schema change while we're still in the process of fleshing things out.